pages/[slug].tsx

```diff
@@ -1,52 +1,59 @@
-import type { ExtendedStory } from '@prezly/sdk/dist/types';
-import { GetServerSideProps, NextPage } from 'next';
-import { BasePageProps } from 'types';
-import { getPrezlyApi } from '@/utils/prezly';
-import Story from '@/modules/Story';
-import Layout from '@/components/Layout';
-import { NewsroomContextProvider } from '@/contexts/newsroom';
-
-interface Props extends BasePageProps {
-    story: ExtendedStory;
-}
+import {
+    DUMMY_DEFAULT_LOCALE,
+    getNewsroomServerSideProps,
+    getPrezlyApi,
+    processRequest,
+    useCurrentStory,
+} from '@prezly/theme-kit-nextjs';
+import type { GetServerSideProps } from 'next';
+import dynamic from 'next/dynamic';
+
+import { importMessages } from '@/utils';
+
+const Story = dynamic(() => import('@/modules/Story'), { ssr: true });
 
-const StoryPage: NextPage<Props> = ({
-    story, categories, newsroom, companyInformation,
-}) => (
-    <NewsroomContextProvider
-        categories={categories}
-        newsroom={newsroom}
-        companyInformation={companyInformation}
-    >
-        <Layout>
-            <Story story={story} />
-        </Layout>
-    </NewsroomContextProvider>
-);
+function StoryPage() {
+    const currentStory = useCurrentStory();
+
+    if (!currentStory) {
+        return null;
+    }
+
+    return <Story story={currentStory} />;
+}
 
 export const getServerSideProps: GetServerSideProps = async (context) => {
     const api = getPrezlyApi(context.req);
+
     const { slug } = context.params as { slug?: string };
     const story = slug ? await api.getStoryBySlug(slug) : null;
 
     if (!story) {
         return { notFound: true };
     }
 
-    const [categories, newsroom, companyInformation] = await Promise.all([
-        api.getCategories(),
-        api.getNewsroom(),
-        api.getCompanyInformation(),
-    ]);
-
-    return {
-        props: {
-            story,
-            categories,
-            newsroom,
-            companyInformation,
+    // We are passing the story object to detect locale from the story itself
+    // since the locale code is not part of the URL (e.g. /my-story).
+    const { serverSideProps } = await getNewsroomServerSideProps(context, { story });
+
+    const { locale } = context;
+    if (locale && locale !== DUMMY_DEFAULT_LOCALE) {
+        return {
+            redirect: {
+                destination: `/${slug}`,
+                permanent: true,
+            },
+        };
+    }
```

Comment:
I wonder if we should move this logic in `processRequest` method of the `theme-kit-nextjs` 🤔 
We're already handling locale redirects there, so it makes sense to also do the story redirects.

Reply:
Maybe, maybe, maybe.. 😄